code/game/machinery/vendors/vendor_crits.dm

```diff
@@ -0,0 +1,115 @@
+	victim.bleed(150)
+	var/obj/item/organ/external/leg/right = victim.get_organ(BODY_ZONE_R_LEG)
+	var/obj/item/organ/external/leg/left = victim.get_organ(BODY_ZONE_L_LEG)
+	if(istype(left))
+		left.receive_damage(200)
+	if(istype(right))
+		right.receive_damage(200)
+
+	if(left || right)
+		victim.visible_message(
+			"<span class='danger'>[victim]'s legs shatter with a sickening crunch!</span>",
+			"<span class='userdanger'>Your legs shatter with a sickening crunch!</span>",
+			"<span class='danger'>You hear a sickening crunch!</span>"
+		)
+
+	return machine.squish_damage - 10
+
+/datum/vendor_crit/pin
+
+/datum/vendor_crit/pin/tip_crit_effect(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	machine.forceMove(get_turf(victim))
+	machine.buckle_mob(victim, force=TRUE)
+	victim.visible_message(
+		"<span class='danger'>[victim] gets pinned underneath [src]!</span>",
+		"<span class='userdanger'>You are pinned down by [src]!</span>"
+	)
+
+	return 0
+
+/datum/vendor_crit/embed
+
+/datum/vendor_crit/embed/is_valid(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	. = ..()
+	if(machine.num_shards <= 0)
+		return FALSE
+
+/datum/vendor_crit/embed/tip_crit_effect(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	victim.visible_message(
+		"<span class='danger'>[src]'s panel shatters against [victim]!</span>",
+		"<span class='userdanger>[src] lands on you, its panel shattering!</span>"
+	)
+
+	for(var/i in 1 to machine.num_shards)
+		var/obj/item/shard/shard = new /obj/item/shard(get_turf(victim))
+		// do a little dance to force the embeds, but make sure the glass isn't gigapowered afterwards
+		shard.embed_chance = 100
+		shard.embedded_pain_chance = 5
+		shard.embedded_impact_pain_multiplier = 1
+		shard.embedded_ignore_throwspeed_threshold = TRUE
+		victim.hitby(shard, skipcatch = TRUE, hitpush = FALSE)
+		shard.embed_chance = initial(shard.embed_chance)
+		shard.embedded_pain_chance = initial(shard.embedded_pain_chance)
+		shard.embedded_impact_pain_multiplier = initial(shard.embedded_pain_multiplier)
+		shard.embedded_ignore_throwspeed_threshold = initial(shard.embedded_ignore_throwspeed_threshold)
+
+	playsound(machine, "shatter", 50)
+
+	return machine.squish_damage - 20
+
+/datum/vendor_crit/pop_head
+
+/datum/vendor_crit/pop_head/tip_crit_effect(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	// pop!
+	var/obj/item/organ/external/head/O = victim.get_organ("head")
+	var/obj/item/organ/internal/brain/B = victim.get_int_organ_tag("brain")
+	if(O)
+		victim.visible_message("<span class='danger'>[O] gets crushed under [src]!</span>", "<span class='userdanger'>Oh f-</span>")
+		O.disfigure()
+		victim.apply_damage(50, BRUTE, BODY_ZONE_HEAD)
+	else
+		O.visible_message("<span class='danger'>[victim]'s head seems to be crushed under the machine...but wait, they had none in the first place!</span>")
+	if(B in O)
+		B.damage += 80
```

Comment:
Is this enough to apply the brain damage? Or does it need another proc to trigger a health update?